posthog/api/organization_member.py

```diff
@@ -0,0 +1,88 @@
+from django.db.models import Model, QuerySet, query
+from django.http.response import Http404
+from django.shortcuts import get_object_or_404
+from rest_framework import exceptions, mixins, response, serializers, status, viewsets
+from rest_framework.permissions import SAFE_METHODS, BasePermission
+from rest_framework.request import Request
+from rest_framework_extensions.mixins import NestedViewSetMixin
+
+from posthog.models import OrganizationMembership
+from posthog.permissions import OrganizationAdminWritePermissions, OrganizationMemberPermissions, extract_organization
+
+
+class OrganizationAdminWriteExcludingDeletePermissions(BasePermission):
+    """Require organization admin level to change object, allowing everyone read AND delete."""
+
+    message = "Your organization access level is insufficient."
+
+    def has_object_permission(self, request: Request, view, object: Model) -> bool:
+        if request.method in SAFE_METHODS or request.method == "DELETE":
```

Comment:
What do you mean by stuff?

Reply:
So the permission validates that anyone can do any safe method (GET, HEAD, OPTIONS) but DELETE too, why allow DELETE is my question

Reply:
Ah, yup, it used to be that `destroy` in the viewset itself checked if the user is only trying to delete themselves (which is allowed), not someone else (which is disallowed) – but good point, move that to this permission class so that it's purpose is clearer.